### PR TITLE
Run3-sim82 Define fine calorimeters thrugh python

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Define fine calorimeters thrugh python

#### PR validation:

Tested using a cfg file from SimG4CMS/Calo/test/python

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: SimG4Core/Application/python/g4SimHits_cfi.py [hunk 1]
```diff
@@ -45,6 +45,9 @@
     DoFineCalo = cms.bool(False),
     # currently unused; left in place for future studies
     EminFineTrack = cms.double(10000.0),
+    FineCaloNames = cms.vstring('ECAL', 'HCAL', 'HGCal', 'HFNoseVol', 'VCAL'),
+    FineCaloLevels = cms.vint32(4, 4, 8, 3, 3),
```
Comment (kpedro88) [inline SimG4Core/Application/python/g4SimHits_cfi.py:49]: is it clear that there will always be only level per subdetector?

**Comment by kpedro88** [comment]
@tklijnsma @kdlong @lgray @intrepid42 FYI

File: SimG4Core/Application/python/g4SimHits_cfi.py (hunk 1 above)
Comment (bsunanda) [reply SimG4Core/Application/python/g4SimHits_cfi.py:49]: Kevin, you are right. They cannot be constants - they depend on geometry implementation. But  that is rare. So I keep them in parameter statements but more stable than the choice of FineCalo in an application

Comment (kpedro88) [reply SimG4Core/Application/python/g4SimHits_cfi.py:49]: That's a good point, but my intended question was slightly different: what if HGCal needs two levels, e.g. 8 and 10? Will that ever happen, or are the sensitive elements only ever at one level for a subdetector?

Comment (bsunanda) [reply SimG4Core/Application/python/g4SimHits_cfi.py:49]: We refer to only top level volumes and not sensitive detector. These top volumes so far happens to be in one level. If not, the same name may happen multiple times with the corresponding levels and then the current logic will work.

**Comment by civanch** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**